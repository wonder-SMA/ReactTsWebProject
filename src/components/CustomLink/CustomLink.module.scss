.component {
  position: relative;
  margin-left: 16px;
  cursor: pointer;

  @media screen and (max-width: 575px) {
    & {
      display: flex;
      align-items: center;
      justify-content: center;
      width: 120px;
      height: 46px;
      margin: 4px 0;
      border: 2px solid white;
      border-radius: 8px;
    }

    @mixin arrow {
      position: absolute;
      font-size: 28px;
      color: white;
    }

    .arrow_left {
      @include arrow;
      left: -20px;
    }

    .arrow_right {
      @include arrow;
      right: -20px;
    }
  }
}
