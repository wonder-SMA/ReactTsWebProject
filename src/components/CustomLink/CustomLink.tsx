import React from 'react';
import { Link, LinkProps, useMatch, useResolvedPath } from 'react-router-dom';

const CustomLink: React.FC<LinkProps> = ({ children, to, ...props }) => {
  let resolved = useResolvedPath(to);
  let match = useMatch({ path: resolved.pathname, end: true });

  return (
    <li>
      <Link
        style={{ borderBottom: match ? "2px white solid" : "none" }}
        to={to}
        {...props}
      >
        {children}
      </Link>
    </li>
  );
}

export default CustomLink;
