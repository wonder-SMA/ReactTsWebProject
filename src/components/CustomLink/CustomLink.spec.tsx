import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';

import CustomLink from './';

describe('CustomLink component', () => {
  it('Should contain text node', () => {
    const { container, getByText } = render(
      <MemoryRouter>
        <CustomLink to="home">Главная</CustomLink>
      </MemoryRouter>
    );
    expect(getByText('Главная')).toBeInTheDocument();
    expect(container.firstChild).toMatchInlineSnapshot(`
      <li
        class="component"
      >
        <a
          href="/home"
        >
          Главная
        </a>
      </li>
    `);
  });
});
