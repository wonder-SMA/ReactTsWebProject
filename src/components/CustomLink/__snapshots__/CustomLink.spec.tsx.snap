// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomLink component Should contain text node 1`] = `
<li>
  <a
    href="/home"
  >
    Главная
  </a>
</li>
`;
